src/core/org/apache/jmeter/report/processor/StatisticsSummaryData.java

```diff
@@ -163,40 +164,8 @@ public final void setMax(long max) {
     /**
      * @return the percentile1
      */
-    public final PercentileAggregator getPercentile1() {
-        return percentile1;
-    }
-
-    /**
-     * Gets the percentile2.
-     *
-     * @return the percentile2
-     */
-    public final PercentileAggregator getPercentile2() {
-        return percentile2;
-    }
-
-    /**
-     * Gets the percentile3.
-     *
-     * @return the percentile3
-     */
-    public final PercentileAggregator getPercentile3() {
-        return percentile3;
-    }
-
-    /**
-     * Instantiates a new statistics info.
-     * @param percentileIndex1 value of first percentile
-     * @param percentileIndex2 value of second percentile
-     * @param percentileIndex3 value of third percentile
-     */
-    public StatisticsSummaryData(long percentileIndex1, long percentileIndex2,
-            long percentileIndex3) {
-        percentile1 = new PercentileAggregator(percentileIndex1);
-        percentile2 = new PercentileAggregator(percentileIndex2);
-        percentile3 = new PercentileAggregator(percentileIndex3);
-        mean = new MeanAggregator();
+    public final long getPercentile(double percent) {
+        return histogram.getValueAtPercentile(percent)/1000000;
```

Comment:
This should not be needed as per https://stackoverflow.com/questions/56986503/time-unit-problem-about-latencyutils-recordlatency-method

Reply:
The only answer in thie question doesn't give us enough information.
So don't we need to muliple and divide by 1,000,000 each time?

Reply:
When the number of Thread is a big number like 1000 -10000, the maximum is bigger than percentile 99%. And IntcountHistogram can also give a correct result.